app/controllers/dish_controller.rb

```diff
@@ -0,0 +1,27 @@
+class DishController < ApplicationController
+  def index
+    @dishes = Dish.all
+  end
+  
+  def show 
+    @dish = Dish.find(params[:id])
+    @dish_calories = @dish.calorie_count
+  end
+
+  def new
+  end
+
+  def update
+    @dish = Dish.find(params[:id])
+    ingredient = Ingredient.find(params[:ingredient_id])
+    
+    @dish.ingredients << ingredient
+    redirect_to @dish
+  end
+  
+  private
+
+  def dish_params
+    params.permit(:name, :description)
+  end
+end
```

Comment:
In the README, I don't believe there was a need for a `index` or `new` view, so I'm wondering why there is implementation for these. I would suggest you take some time to read and understand what you are being asked to implement from the README, and reach out with any questions you have. IC's are already tough enough without wasting time implementing things that are not necessary.